internal/desktop_browser.py

```diff
@@ -485,17 +484,27 @@ def on_start_recording(self, task):
                     while not started and monotonic() < end_time:
                         if os.path.isfile(task['video_file']):
                             video_size = os.path.getsize(task['video_file'])
-                            if initial_size == None:
+                            if initial_size is None:
                                 initial_size = video_size
                             logging.debug("Video file size: %d", video_size)
                             if video_size > initial_size or video_size > 10000:
                                 started = True
                         if not started:
                             time.sleep(0.1)
-                    self.video_capture_running = True
+                    # Check to make sure ffmpeg is still capturing - if Windows gdigrab doesn't stream,
+                    # ffmpeg will complain with "Failed to capture image (error 5)" and
+                    # "Output file #0 does not contain any stream"
+                    # gdigrab will fail to capture if Secure Desktop and UAC are enabled
+                    # or an RDP session is minimized/disconnected/locked
+                    if self.ffmpeg.poll() is None:
+                        logging.debug("Video capture process is still running...")
+                        self.video_capture_running = True
+                    else:
+                        self.video_capture_running = False
+                        logging.exception("ERROR: ffmpeg started but gdigrab did not provide a stream.")
+                        raise RuntimeError('Video could not be captured with this run. Is UAC enabled or RDP minimized/disconnected/locked?')
```

Comment:
Rather than re-raising an exception, if we want to mark the test as failed we should be able to set the error in self.task['error'].